plugins/infinispan80/src/main/java/org/radargun/stream/example/WordCount.java

```diff
@@ -0,0 +1,44 @@
+package org.radargun.stream.example;
+
+import org.infinispan.stream.CacheCollectors;
+import org.radargun.traits.Streamable;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Roman Macor &lt;rmacor@redhat.com&gt;
+ */
+public class WordCount implements Streamable.StreamFunction {
+
+   private Object result;
+
+   @Override
+   public Object apply(Stream stream) {
+      result = stream.map((Serializable & Function<Map.Entry<String, String>, String[]>) e -> e.getValue().split("[\\p{Punct}\\s&&[^'-]]+")).
```

Comment:
The payload here can be reduced significantly by using Infinispan AdvancedExternalizer [1].

This can be done to the object passed in via map, the one to flatMap and the collect argument.

[1] http://infinispan.org/docs/8.1.x/user_guide/user_guide.html#_advanced_externalizers